corehq/apps/translations/tests/test_transifex_integration.py

```diff
@@ -0,0 +1,256 @@
+from django.test import SimpleTestCase
+
+import polib
+import requests_mock
+from unittest.mock import patch
+
+from corehq.apps.translations.integrations.transifex.client import TransifexApiClient
+from corehq.util.test_utils import TestFileMixin
+
+TOKEN = "1234"
+ORGANIZATION_SLUG = "test-organization"
+PROJECT_SLUG = "test-project"
+RESOURCE_SLUG = "test-resource"
+RESOURCE_NAME = "Test Resource"
+
+DATA_PATH = 'corehq/apps/translations/tests/data/transifex/'
+
+
+class TestTransifexApiClient(TestFileMixin, SimpleTestCase):
+
+    @classmethod
+    @requests_mock.Mocker()
+    def setUpClass(cls, mocker):
+        super().setUpClass()
+        cls.mocker = mocker
+        cls.mocker.register_uri(requests_mock.ANY, requests_mock.ANY, text=cls.route_request)
+        cls.tfx_client = TransifexApiClient(TOKEN, ORGANIZATION_SLUG, PROJECT_SLUG)
+
+    def tearDown(self):
+        super().tearDown()
+        self.mocker.reset_mock()
+
+    @classmethod
+    def route_request(cls, request, context):
+        if request.method == 'DELETE':
+            context.status_code = 204
+            return
+        if request.method == 'GET' and 'downloads/' in request.path:
+            return cls._handle_download_redirect(request, context)
+        return cls._get_file(request, 'json')
+
+    @classmethod
+    def _handle_download_redirect(cls, request, context):
+        if request.path.endswith('content'):
+            return cls._get_file(request, 'txt')
```

Comment:
Right. I think its the right to not hit the API, but we should at least have the file get generated by the view, and mock the data the view needs to make the file and at least test the view.

Reply:
I think this method is fine. This is only used by test_request_download_content_for_resource and test_get_translation with the former (if I’m understanding this correctly) really testing the download method of the api class. It doesn’t generate anything - just passes the the resource ids as parameters and expects a response, which the test mocks as a txt file. I don’t really think we need it, since we’re kind of directly testing the api’s method but I don’t it hurts to have. The other one is bit of the same but the sole assertion being between two mocked files is a little flimsy. Again, without hitting the api its just testing that ResourceTranslationsAsyncDownload.download is making the right requests, which I feel is unecessary but also doesn't hurt to have.